### PR TITLE
move methods to container

### LINKED ISSUE #433: Move some `container` package methods to `container.Container`
Why on earth do we do `container.SetCreationTime(cnr, t)` when this could be `cnr.SetCreationTime(t)`? It'd be OK if this was some unrelated package that is doing something to `container.Container` (like `client.SyncContainerWithNetwork`), but it's the same `container` with the same dependencies and things. Almost all of the functions from `container` could be methods making them easier to use.

### PR DESCRIPTION
**Pull request by smallhive**
close #433

### CODE REVIEW

**Review by roman-khimov — COMMENTED** [review]
ApplyNetworkConfig, AssertNetworkConfig, DisableHomomorphicHashing and IsHomomorphicHashingDisabled are not handled and I think they should be.

Also, please move `CalculateIDFromBinary` into `container/id` (likely with some more appropriate name), it doesn't belong here at all.

File: container/container.go [hunk 1]
```diff
@@ -341,35 +341,35 @@ func (x Container) IterateAttributes(f func(key, val string)) {
...
+func (x *Container) SetName(name string) {
+	x.SetAttribute(attributeName, name)
 }
 
 // Name returns container name set using SetName.
 //
 // Zero Container has no name.
-func Name(cnr Container) string {
-	return cnr.Attribute(attributeName)
+func (x *Container) Name() string {
```
[the remark this answers is not in the document]
Comment (smallhive) [reply container/container.go]: done

File: container/container.go [hunk 2]
```diff
@@ -341,35 +341,35 @@ func (x Container) IterateAttributes(f func(key, val string)) {
...
-	cnr.SetAttribute(attributeTimestamp, strconv.FormatInt(t.Unix(), 10))
+func (x *Container) SetCreationTime(t time.Time) {
+	x.SetAttribute(attributeTimestamp, strconv.FormatInt(t.Unix(), 10))
 }
 
 // CreatedAt returns container's creation time set using SetCreationTime.
 //
 // Zero Container has zero timestamp (in seconds).
-func CreatedAt(cnr Container) time.Time {
+func (x *Container) CreatedAt() time.Time {
```
[the remark this answers is not in the document]
Comment (smallhive) [reply container/container.go]: done

File: container/container.go [hunk 3]
```diff
@@ -485,17 +485,17 @@ func CalculateIDFromBinary(dst *cid.ID, cnr []byte) {
...
+func (x *Container) CalculateID(dst *cid.ID) {
+	CalculateIDFromBinary(dst, x.Marshal())
 }
 
 // AssertID checks if the given Container matches its identifier in CAS of the
 // NeoFS containers.
 //
 // See also CalculateID.
-func AssertID(id cid.ID, cnr Container) bool {
+func (x *Container) AssertID(id cid.ID) bool {
```
[the remark this answers is not in the document]
Comment (smallhive) [reply container/container.go]: done

File: container/container.go [hunk 4]
```diff
@@ -459,17 +459,17 @@ func ReadDomain(cnr Container) (res Domain) {
...
 	}
-	return dst.Calculate(signer, cnr.Marshal())
+	return dst.Calculate(signer, x.Marshal())
 }
 
 // VerifySignature verifies Container signature calculated using CalculateSignature.
 // Result means signature correctness.
-func VerifySignature(sig neofscrypto.Signature, cnr Container) bool {
-	return sig.Verify(cnr.Marshal())
+func (x *Container) VerifySignature(sig neofscrypto.Signature) bool {
```
[the remark this answers is not in the document]
Comment (smallhive) [reply container/container.go]: done

File: container/container.go [hunk 5]
```diff
@@ -430,18 +430,18 @@ func (x Domain) Zone() string {
...
+func (x *Container) WriteDomain(domain Domain) {
+	x.SetAttribute(container.SysAttributeName, domain.Name())
+	x.SetAttribute(container.SysAttributeZone, domain.Zone())
 }
 
 // ReadDomain reads Domain from the Container. Returns value with empty name
 // if domain is not specified.
-func ReadDomain(cnr Container) (res Domain) {
-	name := cnr.Attribute(container.SysAttributeName)
+func (x *Container) ReadDomain() (res Domain) {
```
[the remark this answers is not in the document]
Comment (smallhive) [reply container/container.go]: done

File: container/container.go [hunk 6]
```diff
@@ -459,17 +459,17 @@ func ReadDomain(cnr Container) (res Domain) {
 //
 // Returned errors:
 //   - [neofscrypto.ErrIncorrectSigner]
-func CalculateSignature(dst *neofscrypto.Signature, cnr Container, signer neofscrypto.Signer) error {
+func (x *Container) CalculateSignature(dst *neofscrypto.Signature, signer neofscrypto.Signer) error {
```
[the remark this answers is not in the document]
Comment (smallhive) [reply container/container.go]: done

File: container/container.go [hunk 7]
```diff
@@ -485,17 +485,17 @@ func CalculateIDFromBinary(dst *cid.ID, cnr []byte) {
 // CalculateIDFromBinary.
 //
 // See also Container.Marshal, AssertID.
-func CalculateID(dst *cid.ID, cnr Container) {
-	CalculateIDFromBinary(dst, cnr.Marshal())
+func (x *Container) CalculateID(dst *cid.ID) {
```
[the remark this answers is not in the document]
Comment (smallhive) [reply container/container.go]: done

**Review by cthulhu-rider — APPROVED** [review]

**Comment by smallhive** [comment]
Updated missed methods and moved/renamed CalculateIDFromBinary to container/id

**Review by roman-khimov — APPROVED** [review]

**Comment by smallhive** [comment]
Linters are ok

File: container/id/id.go [hunk 8]
```diff
@@ -113,3 +113,11 @@ func (id *ID) DecodeString(s string) error {
 func (id ID) String() string {
 	return id.EncodeToString()
 }
+
+// FromBinary calculates identifier of the binary-encoded container
+// in CAS of the NeoFS containers and writes it into dst.
```
Comment (notimetoname) [inline container/id/id.go:118]: what `dst` is here?

Comment (smallhive) [reply container/id/id.go]: I think, `id` will be right destination. Updated

**Pull request merged**